From b1092a00449842da62e6c78430d8c97d10e8a197 Mon Sep 17 00:00:00 2001
From: Lei Xu <lei@apache.org>
Date: Wed, 20 Jan 2016 14:26:11 -0800
Subject: [PATCH 1113/2848] HDFS-9655. NN should start JVM pause monitor
 before loading fsimage. (John Zhuge via Lei
 (Eddy) Xu)

(cherry picked from commit 2ec438e8f7cd77cb48fd1264781e60a48e331908)

Conflicts:
        hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNode.java

Change-Id: I88cb3b152b298169e2356daf5e6cbdfcab1dcb94
---
 .../org/apache/hadoop/util/JvmPauseMonitor.java    |    1 +
 .../hadoop/hdfs/server/namenode/NameNode.java      |   10 +++++-----
 2 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/JvmPauseMonitor.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/JvmPauseMonitor.java
index 1fe7796..33d46d8 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/JvmPauseMonitor.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/JvmPauseMonitor.java
@@ -174,6 +174,7 @@ public String toString() {
     public void run() {
       StopWatch sw = new StopWatch();
       Map<String, GcTimes> gcTimesBeforeSleep = getGcTimes();
+      LOG.info("Starting JVM pause monitor");
       while (shouldRun) {
         sw.reset().start();
         try {
diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNode.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNode.java
index ad52eba..fbc81d8 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNode.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/NameNode.java
@@ -659,6 +659,10 @@ protected void initialize(Configuration conf) throws IOException {
     NameNode.initMetrics(conf, this.getRole());
     StartupProgressMetrics.register(startupProgress);
 
+    pauseMonitor = new JvmPauseMonitor(conf);
+    pauseMonitor.start();
+    metrics.getJvmMetrics().setPauseMonitor(pauseMonitor);
+
     if (NamenodeRole.NAMENODE == role) {
       startHttpServer(conf);
     }
@@ -678,11 +682,7 @@ protected void initialize(Configuration conf) throws IOException {
       httpServer.setNameNodeAddress(getNameNodeAddress());
       httpServer.setFSImage(getFSImage());
     }
-    
-    pauseMonitor = new JvmPauseMonitor(conf);
-    pauseMonitor.start();
-    metrics.getJvmMetrics().setPauseMonitor(pauseMonitor);
-    
+
     startCommonServices(conf);
   }
   
-- 
1.7.9.5

